### ISSUE TITLE
Add option for coloring lines similar to usernames

### ISSUE THREAD

**Issue by mossarelli** [hexchat/hexchat#1378]
It would be nice to have an option to color and differentiate lines based on the user that wrote them. I've tried different font settings and window sizes, but when lots of people write in the chat at the same time, it gets hard to follow even when I'm fully awake.

I would like to request an option for coloring a user's response based on the color it has in the list but as a faded highlighting behind the text.

If this feature already exists and I've missed it, please redirect me to documentation on such information.
Thank you for your time.

**Comment by Poorchop** [comment]
>but as a faded highlighting behind the text.

I'm not 100% sure I'm understanding what you're suggesting, but that particular suggestion wouldn't work as you're thinking out of the box. You would have to modify your theme to create the faded colors as needed. Are you suggesting that if someone's text is blue, the text background should be light blue? That would be even more illegible.

Everything you're suggesting should already be possible with a very simple script. I already have a script that does something kind of similar, minus the faded background part.

**Comment by Arnavion** [comment]
Faded as in translucent is not doable, but faded as in light is not hard. Set one or more of the 32-99 colors to light ones and override bgcolor to that.

@Poorchop I'm guessing your script does the same? Not sure why you say "minus the faded background part" then...

**Comment by Poorchop** [comment]
@Arnavion I think it just makes channel message text the same color as the username but only for people that you have explicitly specified. It doesn't add any background colors, which is what he is requesting, but that can be scripted too.

**Comment by mossarelli** [comment]
I was thinking about something like this:
![Image showing what OP meant](http://i.imgur.com/D6gmHbQ.jpg)
This image was done entirely through Photoshop and the green color on Test2 is my event that colors "Your message".

**Comment by Poorchop** [comment]
I still have no idea how your picture represents your original suggestion but I just threw some stuff together and created [this script](http://pastie.org/private/nuhsevaem4zmup2h4as7ew) that might be up your alley.

Just install the Python plugin for HexChat and follow the instructions for loading this script. I tested it for about three seconds so it's probably broken - if it is, let me know and I'll maybe try to do something. This script is also copyright Arnavion, probably bviktor and Peter Železný, and also probably about 500 other people too, so copyright everyone excluding me.

**Comment by mossarelli** [comment]
I ran into many problems trying to run Python because I have it under cygwin and a lot of complicated methods.
I'd like to request this feature built in the program without plugins.

**Comment by TingPing** [comment]
There is no need for a script.. just remove the %O's from the various events in Settings > Text events.
